Activist portal header improvements

Vector-Hector opened the pull request:
## Description
This PR improves alignment and style in the activist header, adds profile menu items and a back arrow in the map view.


## Screenshots
<img width="1054" height="941" alt="zetkin-header-map-view" src="https://github.com/user-attachments/assets/6e1497cb-645a-4119-8c19-3b78c5952ee2" />
<img width="1019" height="401" alt="zetkin-header-menu" src="https://github.com/user-attachments/assets/7eb742b2-25af-4028-ad98-af656d06d217" />
<img width="1042" height="592" alt="zetkin-header-main" src="https://github.com/user-attachments/assets/4f026112-feab-4951-a808-90bc6ab31d09" />
<img width="1050" height="935" alt="zetkin-header-project" src="https://github.com/user-attachments/assets/dcd50474-277c-48bf-bdc6-513924392185" />




## Changes
[Add a list of features added/changed, bugs fixed etc]

* Adds logo in /my/ header
* Moves logo+title up in /my/ header
* Add link to /my/ in the home header
* Adds icons to tabs in /my/ header
* Adds menu items to profile menu
* Swaps direction of profile menu (before it overshooted the page to the right)
* Unifies linkage method with https://github.com/zetkin/app.zetkin.org/pull/3122 (the underline in the link to the organization is only shown on hover)
* Adds a home chevron, meaning a link back to /my/ from any non-home page that uses the `ActivistPortlHeader`
* Centers organization logo by centering the logo in `ZUILogo`

## Notes to reviewer

Borrows updated `ZUILink` from https://github.com/zetkin/app.zetkin.org/pull/3122

richardolsson commented:
> And if I remove that back button on top left, would you keep it on other pages such as the event page?

I would not generally have back buttons here, because it's not obvious to many users where they would lead. But I'm interested in hearing your take on this.

Correct me if I'm wrong, but I believe that when you assess these pages you do it mainly as someone who navigates between different pages on the activist portal. But in reality, most people arrive at event pages and project pages not through the activist portal, but via direct links shared in emails and social media. To them, there is no obvious "back".

That's why I prefer another type of navigation, along the lines of the current one where navigating to a project page from an event page (or from a project page to the organization page) is more "direction neutral". That's not to say that the current navigation is perfect. Maybe, as an example, we could consider having a blurb on event pages to say "This event and 13 other events are part of 'The Project Title'".

sefsh commented:
>change the anchor point to get it more to the left

Don't mind that. This is the fault of the avatar being wrapped in a rectangular button and should be addressed separately.

PR seems fine now, thank you for sticking with it!

richardolsson commented on src/zui/components/ZUILink/index.tsx:
```diff
@@ -5,6 +5,11 @@ import { FC } from 'react';
+   * If true, only show underline when hovering
+   */
+  hoverUnderline?: boolean;
+
```
This is a change to the ZUI design system. I just want to double check with @sefsh that this is an intended option for links in ZUI?

richardolsson commented on src/features/organizations/components/ActivistPortlHeader/index.tsx:
```diff
@@ -1,6 +1,7 @@
 import { Box, Button } from '@mui/material';
-import { FC, useState } from 'react';
-import { usePathname, useRouter } from 'next/navigation';
+import { FC, useMemo, useState } from 'react';
```
Putting this comment here towards the top of the file, but it's not specifically about this line.

This file is in a misspelled directory. The name `ActivistPortlHeader` should be `ActivistPortalHeader`. Maybe this is a good opportunity to rename it?

richardolsson reviewed (changes requested):
Nice work! I have tried the preview build and all of the affected pages, as well as reviewed all the changes to the code. I think this is very close to merging, but I did find some minor things that I've commented on below.

I've also seen another issue with the header, and the "Follow" button that is aligned weirdly. I think the intent was to align it towards the end of the row, next to the ellipsis menu. I know this isn't part of this PR, but given the theme of the PR I thought maybe it's a good opportunity to change it. See screenshot below.

Let me know if you prefer that I create a separate issue for it, @Vector-Hector!

<img width="673" height="293" alt="image" src="https://github.com/user-attachments/assets/81e85ae8-3795-4707-bc8a-d8b9b7fe1563" />

Vector-Hector commented on src/zui/components/ZUITabbedNavBar/index.tsx:
```diff
@@ -118,7 +118,7 @@ const ZUITabbedNavBar: FC<ZUITabbedNavBarProps> = ({
             )
           }
-          iconPosition="end"
+          iconPosition={'end'}
```
Yes, it's probably something my auto formatter did. I'll revert the change

Vector-Hector commented:
Alright I adjusted everything except for the ZUILink thing where we need confirmation.

I rebased this onto main to resolve conflicts. 

> I've also seen another issue with the header, and the "Follow" button that is aligned weirdly.

Ziggabyte fixed it in https://github.com/zetkin/app.zetkin.org/commit/cb93f809452b904bafac8c967ee072c71b80f464 This branch just was a few days out of sync. :)

richardolsson reviewed (approved):
Nice! I have reviewed the changes and tried the most recent preview build, and I think we should go ahead and merge!

I will squash and merge since the resulting changes are quite small compared to the number of commits.
